modules/table/src/main/java/org/apache/ignite/internal/table/AbstractTableView.java

```diff
@@ -102,13 +101,13 @@ abstract class AbstractTableView<R> implements CriteriaQuerySource<R> {
     /**
      * Waits for operation completion.
      *
-     * @param fut Future to wait to.
+     * @param future Future to wait to.
      * @param <T> Future result type.
      * @return Future result.
      */
-    protected final <T> T sync(Supplier<CompletableFuture<T>> fut) {
+    protected static <T> T sync(CompletableFuture<T> future) {
         try {
-            return fut.get().get();
+            return future.get();
```

Comment:
I think the idea was that exceptions thrown by the method that creates the future will be handled in a similar way. Are you sure you don't break anything with this change?

Reply:
No, the idea was to have ability to do something before the operation starts being executed; I know this because I introduced that indirection :) This change just reverts it back.